Git usage with single main branch (#34)
infinum/android-handbook: Using Git.md

git merge dev
```

### Release branches

> Most of the projects have one active stream at the moment. This means that all of the work done by developer can go directly into the dev branch. All the work goes into the next release so there is no need to seperate the dev branch into mulitple release branches. 

Reviewer: ```suggestion
Most projects have one active development stream at the moment. This means that all new changes should be included in the next upcoming release. Because of this, all changes can be merged into the `dev` branch without the need to introduce multiple `release` branches.
```

> However, some projects have multiple releases plan ahead. It is possible that at some point a development team will work on multiple releases at the same time. In that case, `release` branches should be introduced. Protect this branches as you do for your main branch. You can use regex to match all the release branches and not worry about protecting a newly create ones. **Note:** If you want to delete a protected branch on Github you will need to remove the protection cause this can't be done even with admin privilege. 

Reviewer: ```suggestion
However, some projects have multiple releases planned ahead. It is possible that at some point a development team will work on multiple releases at the same time. In that case, `release` branches should be introduced and handled in the same manner as the main `dev` branch. You can use regex to match all the release branches and not worry about protecting newly created ones. **Note:** If you want to delete a protected branch on GitHub you will need to remove the protection because this can't be done even with admin privileges. 
```

> To be able to work with release branches we need to have defined version for each specified task. If you are not certain in which version some task is plan to be release ask the project manage before strating the work. In (Productive)[[https://app.productive.io](https://app.productive.io/)] release versions can be defined with `tags` or `task list`. In (JIRA)[<https://www.atlassian.com/software/jira>] you can track the status of each release and define `Fix version/s` attribute on a task.

Reviewer: ```suggestion
When working with `release` branches, it needs to be clear which task should be included in which version. If you are not sure of this, ask the project manager before starting with the task. In (Productive)[[https://app.productive.io](https://app.productive.io/)] release versions can be defined with tags or boards. In (JIRA)[<https://www.atlassian.com/software/jira>] you can track the status of each release and define `Fix version/s` attribute on a task.
```

> When working with release branches make sure you set a tag on the last commit in the release branch. You should do this before merging the code to the dev branch. After publishing and merging release branch to dev, don't forget to also update all other active release branches.  

Reviewer: ```suggestion
After releasing a new version from the `release` branch, make sure to create a new tag with that version from the last commit in that branch. Then you can merge the `release` branch into the `dev` branch. Don't forget to also update all other active `release` branches.
```
